src/composite/MegaChart.js

```diff
@@ -115,6 +117,81 @@
         this._csvButton.click = function (a) {
             context.downloadCSV();
         };
+        
+        this._maximizeButton = new Button()
+            .classed({ "composite_MegaChart-Maximize": true })
+            .id(this.id() + "_maximize")
+            .value("")
+        ;
+        this._maximizeButton.click = function (a) {
+            var target = context.target();
+            var node = target;
+            
+            var isMaximized = target.classList.contains('__hpccisMaximized');
```

Comment:
I would prefer if this was done via d3 (as its method has better browser compatibility for "classLIst"):
```
var targetElement = d3.select(this.target());
var isMaximised = targetElement.classed("__hpccisMaximized");

```